components/browser/state/src/main/java/mozilla/components/browser/state/store/BrowserStore.kt

```diff
@@ -4,24 +4,29 @@
 
 package mozilla.components.browser.state.store
 
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.asCoroutineDispatcher
 import mozilla.components.browser.state.action.BrowserAction
 import mozilla.components.browser.state.reducer.BrowserStateReducer
 import mozilla.components.browser.state.selector.findTab
 import mozilla.components.browser.state.state.BrowserState
 import mozilla.components.lib.state.Action
 import mozilla.components.lib.state.Store
 import java.lang.IllegalArgumentException
+import java.util.concurrent.Executors
 
 /**
  * The [BrowserStore] holds the [BrowserState] (state tree).
  *
  * The only way to change the [BrowserState] inside [BrowserStore] is to dispatch an [Action] on it.
  */
 class BrowserStore(
-    initialState: BrowserState = BrowserState()
+    initialState: BrowserState = BrowserState(),
+    dispatcher: CoroutineDispatcher = Executors.newSingleThreadExecutor().asCoroutineDispatcher()
```

Comment:
We shouldn't need to expose the `dispatcher` here. Is this for testing purposes?

Reply:
It is, there might be a better solution I'm missing though.  `SearchFeature` responds to `UpdateSearchRequestAction` then dispatches a `ConsumeSearchRequestAction` in its collect block.  AFAICT, using `joinBlocking` [here](https://github.com/mozilla-mobile/android-components/pull/5669/files#diff-3bde79026c2bd8586e7ab86d45f42a39R102-R113) doesn't cause the consume action to block, leading to a flaky test.  It's possible I'm misunderstanding something though?

Reply:
I'll try your suggestion in https://github.com/mozilla-mobile/android-components/pull/5669#discussion_r378431112

Reply:
@Baron-Severin ah we have an extension function you can use in tests for this (to make sure actions have been processed): https://github.com/mozilla-mobile/android-components/blob/f7d2a4648162bebc98f852e2f2f9d6bd8b6f0636/components/feature/prompts/src/test/java/mozilla/components/feature/prompts/PromptFeatureTest.kt#L218